lib/scenic/theme.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Scenic.Theme do
+  @moduledoc """
+  A Theme is a collection of settings for things like font, color pallettes, and so forth.
+  """
+  defstruct font: nil, pallette: %{}, name: nil
+
+  @type pallette :: %{
+          atom() => Scenic.Color.t()
+        }
```

Comment:
We assume that a pallette is something like:

```elixir
%{
primary: Scenic.Color.named().red(),
secondary: Scenic.Color.named().yellow()
}
```
I wonder if we want to scope-creep this to be any sort of fill or pattern?

Reply:
Given the types here we can also just use simple atom names for the colors (via the implicit color mechanism) in `Scenic.Color`

```elixir
%{
primary: :red,
secondary: :yellow
}
```

Reply:
I would like the type of a color to be Scenic.Color.t(), which allows for either format of color.
```
%{
primary: :red,
secondary: :yellow
}
```
is a very convenient shorthand.
But sometimes you want to be explicit, or add an alpha channel, so that needs to be supported too.